apps/app_record.c

```diff
@@ -514,6 +518,9 @@ static int record_exec(struct ast_channel *chan, const char *data)
 		ast_channel_stop_silence_generator(chan, silgen);
 
 out:
+	snprintf(durationbuf, sizeof(durationbuf), "%ld", ast_tvdiff_ms(ast_tvnow(), start));
+	pbx_builtin_setvar_helper(chan, "RECORD_TIME", durationbuf);
```

Comment:
I think in all cases RECORD_TIME should be set, in error cases it should be set to 0 so that there isn't state carryover between invocations, and it should be documented as such in the variable XML.